ingestion/src/metadata/ingestion/source/dashboard/powerbi/metadata.py

```diff
@@ -539,6 +552,90 @@ def create_datamodel_report_lineage(
                 )
             )
 
+    def _get_table_and_datamodel_lineage(
+        self,
+        db_service_name: str,
+        table: PowerBiTable,
+        datamodel_entity: DashboardDataModel,
+    ) -> Optional[Iterable[Either[AddLineageRequest]]]:
+        """
+        Method to create lineage between table and datamodels
+        """
+        try:
+            table_fqn = fqn.build(
+                self.metadata,
+                entity_type=Table,
+                service_name=db_service_name,
+                database_name=None,
+                schema_name=None,
+                table_name=table.name,
+            )
+            table_entity = self.metadata.get_by_name(
+                entity=Table,
+                fqn=table_fqn,
+            )
```

Comment:
Is there any way to defer this API call to the Sink? We are avoiding having API calls on the Source due to how Multithreading works.

It should be ok if this is ran as a post_process step of the root node but it'd be better if we could just defer the call to the Sink like we did with the OMetaLifecycle